crux-mir-comp/DESIGN.md

```diff
@@ -19,7 +19,7 @@ fn f(x: (u8, u8)) -> (u8, u8) {
     (x.1, x.0)
 }
 
-#[crux_test]
+#[cfg_attr(crux, crux::test)]
```

Comment:
```suggestion
#[crux::test]
```
`cfg_attr(crux, ...)` doesn't make sense here because compiling the function outside crux-mir will fail for other reasons (it uses `crucible_assume!`, for example).

Reply:
Ah, that was a result of me blindly copy-pasting things. I suppose my uses of `cfg_attr` in the `crux-mir-comp` test suite are _also_ wrong as a result of this.

Reply:
Yes - in general `#[cfg_attr(crux, crux::test)]` is a strange thing to do, since under `cfg(not(crux))` it can't use `symbolic` etc, but under `cfg(crux)` it doesn't make sense to have a `#[crux::test]` with no calls to `symbolic`.

The one place `#[cfg_attr(crux, crux::test)]` makes sense is in the crux-mir conc_eval tests, which are a special case.  (Though apparently we were also using it in symb_eval, presumably due to a similar copy-paste issue.)

Reply:
We now consistently use `#[crux::test]` in test cases that use the `crucible` crate.